server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -235,36 +238,91 @@ protected void doExecute(Task task, BulkRequest bulkRequest, ActionListener<Bulk
             if (autoCreateIndices.isEmpty()) {
                 executeBulk(task, bulkRequest, startTime, listener, responses, indicesThatCannotBeCreated);
             } else {
-                final AtomicInteger counter = new AtomicInteger(autoCreateIndices.size());
-                for (String index : autoCreateIndices) {
-                    createIndex(index, bulkRequest.preferV2Templates(), bulkRequest.timeout(), new ActionListener<>() {
-                        @Override
-                        public void onResponse(CreateIndexResponse result) {
-                            if (counter.decrementAndGet() == 0) {
-                                threadPool.executor(ThreadPool.Names.WRITE).execute(
-                                    () -> executeBulk(task, bulkRequest, startTime, listener, responses, indicesThatCannotBeCreated));
+                if (state.getNodes().getMinNodeVersion().onOrAfter(Version.V_8_0_0)) {
+                    CheckedConsumer<AutoCreateAction.Response, Exception> handler = response -> {
+                        Exception suppressed = null;
+                        for (Map.Entry<String, Exception> entry : response.getFailureByNames().entrySet()) {
+                            Exception e = entry.getValue();
+                            if (e == null || ExceptionsHelper.unwrapCause(e) instanceof ResourceAlreadyExistsException) {
```

Comment:
I think it would be nicer to handle the `ResourceAlreadyExistsException` in the auto-create action.